### ISSUE TITLE
Doesn't extract styles from Figma Components

### ISSUE THREAD

**Issue by davidyeiser** [jxnblk/figma-theme#2]
This is based on my very limited testing…

Take this file in Figma, just a box with the Style 'Red' applied:
<img width="961" alt="screenshot 2018-07-09 17 04 52" src="https://user-images.githubusercontent.com/175191/42475989-6b8ff80e-839a-11e8-9046-44ca650e3814.png">

The script output is like so:

````
{
  "colors": {
    "Red": "#eb5757"
  },
  "textStyles": {},
  "fonts": [],
  "fontSizes": [],
  "fontWeights": [],
  "lineHeights": []
}
````

However, if you convert it to a Figma Component:
<img width="961" alt="screenshot 2018-07-09 17 05 55" src="https://user-images.githubusercontent.com/175191/42475996-74643670-839a-11e8-85c5-244f0c407d0c.png">

And re-run the script, the style disappears:

````
{
  "colors": {},
  "textStyles": {},
  "fonts": [],
  "fontSizes": [],
  "fontWeights": [],
  "lineHeights": []
}
````

**Comment by davidyeiser** [comment]
Follow up: when I do two boxes, one a Component and one not, it grabs the Style only from the non-component. 

Also I get no error messages or anything, just this:

````
[figma-theme] fetching data for: <file ID>
[figma-theme] parsing data...
[figma-theme] file saved /path/to/theme.json
````

**Comment by jxnblk** [comment]
Thanks! Would you mind sharing a link to a Figma project I can test this with?

**Comment by davidyeiser** [comment]
Yep, np: https://www.figma.com/file/Yw9L6FATzLpdcsnA5vdSgCRT/figma-theme-Test

**Comment by davidyeiser** [comment]
Looking at the raw API returned from Figma, when an object is converted to a Component, it get moves to a `children` array along with its `styles` object.

For the two rectangles in that example Figma file, the rectangle object is returned like this:

````
{
  id: "2:0"
  name: "box"
  type: "RECTANGLE"
  …
  styles: { … }
}
````

Whereas the component is returned like this (no `styles` in the main object):

````
{
  id: "4:0"
  name: "box"
  type: "COMPONENT"
  …
  children: [{
    id: "2:9"
    name: "box"
    type: "RECTANGLE"
    …
    styles: { … }
  }]
}
````

**Comment by jxnblk** [comment]
Thanks! It should be flattening the `children` arrays here, but I'll take a close look in a bit: https://github.com/jxnblk/figma-theme/blob/master/index.js#L55

**Comment by jxnblk** [comment]
Published `v1.0.0-3` with a fix, but please reopen the issue if it still doesn't work on your end

**Comment by davidyeiser** [comment]
Yep that fixed it, thanks!
